bats/tests/containers/image-allow-list.bats

```diff
@@ -0,0 +1,193 @@
+setup() {
+    load '../helpers/load'
+    RD_USE_IMAGE_ALLOW_LIST=true
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine --kubernetes.enabled=true --kubernetes.version="$RD_KUBERNETES_PREV_VERSION"
+    wait_for_apiserver "$RD_KUBERNETES_PREV_VERSION"
+    wait_for_container_engine
+}
+
+@test 'update the list of patterns first time' {
+    # Can't use bats run with a heredoc so write to a temp file
+    TEMPFILE=$(mktemp -t RD_TEST_63.XXXXX)
+    cat >"$TEMPFILE" <<EOF
+{
+  "version": 6,
+  "containerEngine": {
+    "allowedImages": {
+      "enabled": true,
+      "patterns": ["nginx", "busybox", "python"]
+    }
+  }
+}
+EOF
+    run rdctl api settings -X PUT --input "$TEMPFILE"
+    assert_success
+    rm -f "$TEMPFILE"
```

Comment:
Why do you need to use `run`? You don't capture the output, and `assert_success`, so you can just not use `run`, and rely on BATS failing the test if the command fails:

```suggestion
    run rdctl api settings -X PUT --input <<EOF
{
  "version": 6,
  "containerEngine": {
    "allowedImages": {
      "enabled": true,
      "patterns": ["nginx", "busybox", "python"]
    }
  }
}
EOF
```

You also run this code often enough that you should create a function for it, e.g.

```
set_allowed_images '"nginx", "busybox", "python"'
```

Unless you have an idea to make this work (in a simple, readable manner):

```
set_allowed_images nginx busybox python
```

Reply:
I used tempfiles because I didn't realize that `run CMD; assert_success` is the same as `CMD` and `shellcheck` wanted to see an assertion on the line after a `run`.

Sometimes `enabled` is set to false` instead of `true`, not sure there's much to be gained by trying to refactor all of that into a function.